### ISSUE TITLE
In the docker image, the Chinese character appears as rectangle

### ISSUE THREAD

**Issue by tonyjonson8748** [ScottPlot/ScottPlot#4284]
**Question:** <br>
Hi brother!<br>
When I run the official Demo https://scottplot.net/cookbook/5.0/Internationalization/AutomaticFontDetection in Docker(everything works fine on Mac and Windows), all non-English text is displayed as rectangle.
When I found this problem, I also refer to the official articles at https://scottplot.net/faq/linux-font/ to add some package in `Dockerfile`.<br>
I also refer to #2746 .So I execute the test code :
```cs
var familyName= SKFontManager.Default.MatchCharacter('汉').FamilyName;
``` 
it will throw an exception:
```
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
```
I guess if the Docker image does not have the corresponding font library, causing an error. I am very much looking forward to your reply and wish you a good day！

![demo](https://github.com/user-attachments/assets/463b6630-9609-4914-89e0-fec12b3dbe2e)

**Dotnet Version:** dotnet 6.0

**Dockerfile:** 
```dockerfile
FROM  mcr.microsoft.com/dotnet/aspnet:6.0
WORKDIR /app
ENV TZ=Asia/Shanghai
RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
RUN apt-get update
RUN apt-get install -y libfreetype6
RUN apt-get install -y libfontconfig1
RUN apt-get install -y fontconfig

COPY ./publish /app
CMD ["dotnet", "DrawImage.dll"]
```

**ScottPlot Version:** 5.0.39

**Code Sample:** 

```cs
using ScottPlot;
using SkiaSharp;

Plot myPlot = new();

var sig1 = myPlot.Add.Signal(Generate.Sin(phase: .1));
var sig2 = myPlot.Add.Signal(Generate.Sin(phase: .2));
var sig3 = myPlot.Add.Signal(Generate.Sin(phase: .3));

sig1.LegendText = "测试"; // Chinese
sig2.LegendText = "試験"; // Japanese
sig3.LegendText = "테스트"; // Korean
myPlot.ShowLegend();

myPlot.Title("测试"); // Chinese
myPlot.YLabel("試験"); // Japanese
myPlot.XLabel("테스트"); // Korean

myPlot.Font.Automatic(); // set font for each item based on its content

myPlot.SavePng("demo.png", 400, 300);
var familyName= SKFontManager.Default.MatchCharacter('汉').FamilyName;
//when I execute this code
// SKFontManager.Default.MatchCharacter('汉').FamilyName
//,it will throw an exception:
//Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
```

**Comment by LeaFrock** [comment]
It should be the problem of your docker config which lacks chinese fonts (and maybe this [cnblog](https://www.cnblogs.com/bill89/p/13987266.html) can help), rather than this library.

**Comment by VibrationAnalystCN** [comment]
一看开头的“Hi，brother！”就是中国老铁
![130626BA](https://github.com/user-attachments/assets/628a3318-09ca-4a24-80cb-4957e1141957)

你可以试试指定字体，比如指定“等线”：**.Plot.Font.Set("等线");**。
![等线](https://github.com/user-attachments/assets/e11886d5-8608-4f91-8549-cd17e56a02b4)

**Comment by tonyjonson8748** [comment]
> 一看开头的“Hi，brother！”就是中国老铁 ![130626BA](https://private-user-images.githubusercontent.com/165245736/370098345-628a3318-09ca-4a24-80cb-4957e1141957.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcxNDc2MzgsIm5iZiI6MTcyNzE0NzMzOCwicGF0aCI6Ii8xNjUyNDU3MzYvMzcwMDk4MzQ1LTYyOGEzMzE4LTA5Y2EtNGEyNC04MGNiLTQ5NTdlMTE0MTk1Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyNFQwMzA4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jYjc0ZTgxMjFmY2EwOWQyYzA1NDEwZjMzMmY0YzAxYWU4ZTFiNjMzYWE2MTNiYTMwYjlmMDlkZWNkZGI2NWRhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.LIKLhm_afsdW_mwobKOgUCeHQGrOJgMfvier6_L4Tg8)
> 
> 你可以试试指定字体，比如指定“等线”：**.Plot.Font.Set("等线");**。 ![等线](https://private-user-images.githubusercontent.com/165245736/370098200-e11886d5-8608-4f91-8549-cd17e56a02b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcxNDc2MzgsIm5iZiI6MTcyNzE0NzMzOCwicGF0aCI6Ii8xNjUyNDU3MzYvMzcwMDk4MjAwLWUxMTg4NmQ1LTg2MDgtNGY5MS04NTQ5LWNkMTdlNTZhMDJiNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyNFQwMzA4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yZjI3OGJhOTBkOGJlOTFkN2Q1MWUxNzY3MDU4NzRhZjlmY2M1NzA4ZmE4ZGNmNmZhYjk4Y2FlN2UyZjBlMzMyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ralszlcmmd_s2sUZb5M-8jDA3SAeYY7TQ0DGUbks0wQ)

Thank you, I did my best with my broken English questions（谢谢你，我已经尽力我的蹩脚英文提问了，😀）

**Comment by tonyjonson8748** [comment]
> It should be the problem of your docker config which lacks chinese fonts (and maybe this [cnblog](https://www.cnblogs.com/bill89/p/13987266.html) can help), rather than this library.

Thank you. By following your example, my problem is solved. Here are my solutions:

1. Since the official image does not provide Chinese fonts, I copied the corresponding fonts from my Chinese version of ubuntu operating system into the project.
2. Copies the font to the specified location in the Dockerfile--`COPY ./fonts/* /usr/share/fonts/truetype/`
3. Install the appropriate font configuration library and perform the font configuration
   ```dockerfile
   RUN apt-get install -y fontconfig
   RUN fc-cache -vf
   ```
4. I found that on Linux systems, ScottPlot provides the method `myPlot.Font.Automatic();` can't correct detection font, and also `SKFontManager. Default. MatchCharacter (" 汉"); `  is not recognized correctly.So `myplot.font-set ("xxx") ` is require to hardcode the Family Name of the font.
